thetis/log.py

```diff
@@ -11,38 +11,136 @@
 """
 from thetis.utility import COMM_WORLD
 import logging
+from logging import DEBUG, INFO, WARNING, ERROR, CRITICAL
+import os
 
 __all__ = ('logger', 'output_logger',
-           'debug', 'info', 'warning', 'error', 'critical', 'print_output')
+           'DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL',
+           'log', 'debug', 'info', 'warning', 'error', 'critical',
+           'print_output', 'set_thetis_loggers', 'thetis_log_level',
+           'set_log_directory')
 
+logger_format = {
+    'thetis': '%(name)s:%(levelname)s %(message)s',
+    'thetis_output': '%(message)s',
+}
 
-def get_new_logger(name, fmt='%(levelname)s %(message)s'):
-    logger = logging.getLogger(name)
 
-    for handler in logger.handlers:
-        logger.removeHandler(handler)
+class ThetisLogConfig:
+    """Module-wide config object"""
+    filename = None
 
-    handler = logging.StreamHandler()
-    handler.setFormatter(logging.Formatter(fmt))
 
-    if COMM_WORLD.rank != 0:
-        handler = logging.NullHandler()
+thetis_log_config = ThetisLogConfig()
 
-    logger.addHandler(handler)
-    return logger
 
+def set_thetis_loggers(comm=COMM_WORLD):
+    """Set stream handlers for log messages.
 
-# create conventional logger for error etc messages
-logger = get_new_logger('thetis')
-logger.setLevel(logging.DEBUG)
+    :kwarg comm: The communicator the handler should be collective
+         over. If provided, only rank-0 on that communicator will
```

Comment:
Yes, I copied these docstrings from firedrake's log. Actually here I'm not supporting the `comm=None` case which would allow all procs to log. I haven't really needed this feature.

Reply:
Yeah makes sense - I didn't immediately see how to fix it if they do have to share a logfile